### DISCUSSION TITLE
Restrict Cross-Tenant Login and Disable Loading Page for Unauthorised Users - Sunbird SaaS Release 5.0.0

### DISCUSSION THREAD

**Discussion by siddhishinde0723** [Sunbird-Ed/Community#727] [General Discussion]
Hello Team,

I am using Sunbird SaaS release-5.0.0 APK build. My application is configured for tenant-based login, allowing only users from a specific tenant to log in. However, I am encountering issues with tenant restrictions and loading page behaviour.

**Issue Details:**

1. Cross-Tenant Login Restriction:
Problem: If I do not clear the app's cache, users from different tenants can log in, which should not be allowed.
Expected Behavior: The application should restrict login attempts from users belonging to different tenants without the need to clear the cache.

2. Loading Page Display:
Problem: When a user from a different tenant attempts to log in, the loading page is shown before displaying coming back to login page.
Expected Behavior: The application should directly display a message indicating that the user cannot log in without showing the loading page.

**Attachments:**
For reference, I have attached a video demonstrating the issues.

https://github.com/Sunbird-Ed/Community/assets/72926048/af7c531b-2222-433a-a787-5887bf3f6d47

Any guidance or assistance on how to implement these changes would be greatly appreciated. Thank you!

cc:  @rajeevsathish @mohitga @vinukumar-vs @ashokreddy1208 @swayangjit

**Comment by swayangjit** [comment]
@siddhishinde0723 
**Cross-Tenant Login Restriction:**
Sunbird doesn't have the Cross-Tenant login restriction. As Sunbird is generalized to login any user this can be enhanced to put up restrictions for specific users in the portal backend. 

**Loading Page Display**
The loading page behavior can  be implemented in client side . In this [method](https://github.com/Sunbird-Ed/SunbirdEd-mobile-app/blob/46522167f68066a952fec2e73be982273ce32249/src/services/login-navigation-handler.service.ts#L164C13-L164C30) you can get the tenant id and based on the current tenant id you can manipulate the UI behaviour.

**Reply by siddhishinde0723** [reply]
@swayangjit As you mentioned, i have added the condition and its working. 
But now the issue is if the user is of other tenant then he is not getting the login page need help for that.

**Reply by siddhishinde0723** [reply]
@swayangjit this issue is fixed.
Added route navigation to sign in page when i get error.
